src/django_otp/plugins/otp_email/models.py

```diff
@@ -19,30 +15,18 @@ def key_validator(value):
     return hex_validator()(value)
 
 
-class EmailDevice(Device):
+class EmailDevice(SideChannelDevice):
     """
-    A :class:`~django_otp.models.Device` that delivers a token to the user's
+    A :class:`~django_otp.models.SideChannelDevice` that delivers a token to the user's
     registered email address (``user.email``). This is intended for
     demonstration purposes; if you allow users to reset their passwords via
     email, then this provides no security benefits.
-
-    .. attribute:: key
-
-        *CharField*: A hex-encoded secret key of up to 40 bytes. (Default: 20
-        random bytes)
     """
-    key = models.CharField(max_length=80,
-                           validators=[key_validator],
-                           default=default_key,
-                           help_text='A hex-encoded secret key of up to 20 bytes.')
-
-    @property
-    def bin_key(self):
-        return unhexlify(self.key.encode())
 
     def generate_challenge(self):
-        token = totp(self.bin_key)
-        body = render_to_string('otp/email/token.txt', {'token': token})
+        self.generate_token()
```

Comment:
This whole thing came out of a request for customizing the valid time of email tokens. Feel free to fold in a `OTP_EMAIL_TOKEN_VALID_TIME` or similar setting here if you like.

Reply:
Thanks, I've added a new setting called `OTP_EMAIL_TOKEN_VALID_SECS`. I chose `SECS` instead of `TIME` to make the time unit unambiguous.